web_poet/exceptions/http.py

```diff
@@ -6,14 +6,61 @@
 operations.
 """
 
+from web_poet.page_inputs.http import HttpResponse, HttpRequest
 
-class HttpRequestError(IOError):
-    """Indicates that an exception has occurred when the HTTP Request was being
-    handled.
 
-    For responses that are successful but don't have a ``200`` **status code**,
-    this exception shouldn't be raised at all. Instead, the :class:`~.HttpResponse`
-    should simply reflect the response contents as is.
+class HttpError(IOError):
+    """Indicates that an exception has occurred when handling an HTTP operation.
+
+    This is used as a **base class** for more specific errors and could be vague
+    since it could denote problems either in the HTTP Request or Response.
+
+    For more specific errors, it would be better to use :class:`.HttpRequestError`
+    and :class:`.HttpResponseError`.
+
+    :param request: The :class:`~.HttpRequest` instance that was used.
+    :type request: HttpRequest
+    """
+
+    def __init__(self, *args, request: HttpRequest = None, **kwargs):
```

Comment:
hey! What do you think about simplifying the parameters - why are *args and **kwargs needed here?

Reply:
Good call. Addressed this in https://github.com/scrapinghub/web-poet/pull/38/commits/8b01cf7590f904047dba1851fb4d0ff18a43da69.